### PR TITLE
COMMERCE-13157 - Stack Overflow when merging guest order after logging in

### PR DESCRIPTION
**Pull request by victorcdp**
Relevant Issue: [COMMERCE-13157](https://liferay.atlassian.net/browse/COMMERCE-13157)

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:commerce

File: modules/test/playwright/helpers/HeadlessAdminUserApiHelper.ts [hunk 1]
```diff
@@ -19,4 +19,82 @@ export class HeadlessAdminUserApiHelper {
...
+		const postUserAccount = await this.apiHelpers.post(
+			`${this.apiHelpers.baseUrl}${this.basePath}/user-accounts`,
+			{
+				alternateName: alternateName,
+				currentPassword: currentPassword,
+				emailAddress: emailAddress,
+				familyName: familyName,
+				givenName: givenName,
+				password: password,
+    		}
```
Comment (brianikim) [inline modules/test/playwright/helpers/HeadlessAdminUserApiHelper.ts]: Make sure to provide only the bare minimum for all the API calls.

For example, if the POST can be done with only password and givenName (only an example, probably not true), then you should only use those two properties.

Comment (brianikim) [reply modules/test/playwright/helpers/HeadlessAdminUserApiHelper.ts]: **ALL** your post calls need to have the ability to do the post call with default parameters. See one of my classes for example from LPD-4173

File: modules/test/playwright/tests/commerce/commerceMergeOrdersGUI.spec.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	await page.frameLocator('iframe[title="Permissions"]').locator('#guest_ACTION_VIEW').check();
+	await page.frameLocator('iframe[title="Permissions"]').getByRole('button', { name: 'Save' }).click();
+	await page.getByLabel('close', { exact: true }).click();
+	await page.getByRole('button', { name: 'Pending Orders' }).click();
+	await page.locator('li').filter({ hasText: 'CheckoutCheckoutWidget Page' }).getByRole('button').nth(2).click();
+	await page.getByRole('menuitem', { name: 'Permissions' }).click();
+	await page.frameLocator('iframe[title="Permissions"]').locator('#guest_ACTION_VIEW').check();
+	await page.frameLocator('iframe[title="Permissions"]').getByRole('button', { name: 'Save' }).click();
+	await page.getByLabel('close', { exact: true }).click();
+
```
Comment (brianikim) [inline modules/test/playwright/tests/commerce/commerceMergeOrdersGUI.spec.ts:64]: **ALL of these locators** need to go into a class in pages/commerce. Please see my ticket for example

**Pull request closed without merging**